gptcache/manager/vector_data/pgvector.py

```diff
@@ -0,0 +1,173 @@
+from typing import List
+import numpy as np
+
+from gptcache.manager.vector_data.base import VectorBase, VectorData
+from gptcache.utils import import_sqlalchemy
+
+import_sqlalchemy()
+
+# pylint: disable=wrong-import-position
+from sqlalchemy import create_engine, Column, Index, text  # pylint: disable=C0413
+from sqlalchemy.types import (  # pylint: disable=C0413
+    Integer,
+    UserDefinedType
+)
+from sqlalchemy.orm import sessionmaker  # pylint: disable=C0413
+from sqlalchemy.ext.declarative import declarative_base  # pylint: disable=C0413
+
+Base = declarative_base()
+
+class VectorType(UserDefinedType):
+    """
+    pgvector type mapping for sqlalchemy
+    """
+    cache_ok = True
+
+    def __init__(self, precision = 8):
+        self.precision = precision
+
+    def get_col_spec(self, **_):
+        return f"vector({self.precision})"
+
+    def bind_processor(self, dialect):
+        def process(value):
+            return value
+        return process
+
+    def result_processor(self, dialect, coltype):
+        def process(value):
+            return value
+        return process
+
+def get_model_and_index(table_prefix, vector_dimension, index_type, lists):
+    class VectorStoreTable(Base):
+        """
+        vector store table
+        """
+
+        __tablename__ = table_prefix + "_vector_store"
```

Comment:
`_pg_vector_store` is better, because there will be other vector store using the `sqlalchemy`